### ISSUE TITLE
Track which DataFrame Column corresponds to which Array Column(s) after Transform

### ISSUE THREAD

**Issue by sveitser** [paulgb/sklearn-pandas#13]
I think it would be useful for feature selection if it was possible to keep track of which DataFrame columns were mapped to which array columns during the transformation so that one could use for instance the `feature_importances_` of ensemble methods in sklearn.

Is there a straight-forward way to do this right now? I looked into it a bit but didn't find a common way to get the necessary information during fitting of the sklearn transforms. Therefore the best way I can currently think of is to do the inspection separately for each sklearn transform, i. e.  use `self.feature_names_` for `DictVectorizer`, `self.classes_` for `LabelBinarizer`, etc. 

I'm thinking there must be a better way to do this.

**Comment by dukebody** [comment]
I think it is possible to track at least which columns of the final matrix correspond to each variable. Since the results of the transformation of each set of columns is hstacked in the end, we could keep track of the columns that resulted from the transformation of each variable in a "feature_indices_" variable in the mapper after transformation.

The meaning could be exactly the same one as OneHotEncoder:

    feature_indices_ : array of shape (n_features,)
    Indices to feature ranges. Feature i in the original data is mapped to features from feature_indices_[i] to feature_indices_[i+1] (and then potentially masked by active_features_ afterwards)

@sveitser could you implement that?

**Comment by Motorrat** [comment]
I know there is work done on this issue so just wanted to vote for it. Here is an example of a boilerplate that could be used for many classification jobs driven from a database table. It would be very handy to be able to track back what features (columns) made it to the final set.

```
    #X and y are created from a dataframe
    labels = list(X.select_dtypes(include=['object']).columns)
    booleans = list(X.select_dtypes(include=['bool']).columns)
    integers = list(X.select_dtypes(include=['int64']).columns)
    floats = list(X.select_dtypes(include=['float64']).columns)
 
    mapping=[(b,None) for b in booleans]
    mapping.extend((f, StandardScaler()) for f in floats)
    mapping.extend((i, StandardScaler()) for i in integers) # or LabelBinarizer
    mapping.extend((l,LabelBinarizer()) for l in labels)

    mapper = DataFrameMapper(mapping)

    # the below is illustrative
    clf_pipeline = Pipeline([('map',mapper),('feature_selection', SelectKBest(f_classif, k=5) ),('clf',SGDClassifier(class_weight='balanced'))])
    y_pred = cross_validation.cross_val_predict(clf_pipeline,X,y,cv=8,n_jobs=-1)
    print("Precision: %1.2f, Recall: %1.2f, F1: %1.2f" %
     (precision_score(y, y_pred),recall_score(y, y_pred),f1_score(y, y_pred)))
```

Via clf_pipeline.named_steps['feature_selection'].get_support() we can see what was selected via KBest, but there is no way to track it back to the original X.

**Comment by Motorrat** [comment]
```
this is my workaround
def binarize_label_columns(X):
    numericals = list(X.select_dtypes(include=['int64','float64']).columns)
    labels = list(X.select_dtypes(include=['object']).columns)
    #keep numericals and append binarized labels with new column names
    #we do it outside the pipeline because LabelBinarizer doesn't keep the
    #new column names (classes) for the column lineage
    N=X[numericals] 
    for l in labels:
        mlb=LabelBinarizer()
        L=pd.DataFrame(mlb.fit_transform(X[l]))
        L.columns=[l+"_"+v for v in mlb.classes_]
        N=pd.concat([N,L],axis=1)
    return N
```

**Comment by dukebody** [comment]
Good news! This functionality was addressed in https://github.com/paulgb/sklearn-pandas/commit/2fc6286e2d602a1e1b7d0eab7c256f006eca58ac
